Compilation Error

I cannot test your DB.
I have type this and 
    rustc +nightly --version
    rustc 1.84.0-nightly (e92993dbb 2024-10-18)
And its showing me three 3 errors.
   1 -> error[E0308]: mismatched types
             --> src\paging\cache.rs:400:31
              |
          400 |                 .get_many_mut(frame_ids.map(|frame_id| *frame_id))
              |                  ------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `[usize; _]`, found `Option<usize>`
              |                  |
              |                  arguments to this method are incorrect
              |
              = note: expected array `[usize; _]`
                          found enum `Option<usize>`
          note: method defined here
             --> /rustc/e92993dbb43f0a5d17fe56e2d82f90435d6521c8\library\core\src\slice\mod.rs:4518:12

  2 -> error[E0308]: mismatched types
             --> src\paging\cache.rs:398:9
              |
          397 |       ) -> Option<[&mut MemPage; N]> {
              |            ------------------------- expected `Option<[&mut MemPage; N]>` because of return type
          398 | /         self.pages.get_many_mut(pages.each_ref()).map(|frame_ids| {
          399 | |             self.buffer
          400 | |                 .get_many_mut(frame_ids.map(|frame_id| *frame_id))
          401 | |                 .unwrap()
          ...   |
          405 | |                 })
          406 | |         })
              | |__________^ expected `Option<[&mut MemPage; N]>`, found `[[&mut MemPage; _]; N]`
              |
              = note: expected enum `Option<[&mut MemPage; N]>`
                        found array `[[&mut MemPage; _]; N]`

  3 -> error[E0658]: use of unstable library feature 'pointer_is_aligned_to'
           --> src\storage\page.rs:346:21
            |
        346 |             pointer.is_aligned_to(CELL_ALIGNMENT),
            |                     ^^^^^^^^^^^^^
            |
            = note: see issue #96284 <https://github.com/rust-lang/rust/issues/96284> for more information
            = help: add `#![feature(pointer_is_aligned_to)]` to the crate attributes to enable
            = note: this compiler was built on 2024-10-18; consider upgrading it if it is out of date


-------------------------------------------------Please fix the issues--------------------------------------------------------------

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,6 +1,5 @@
 //! MKDB toy database.
 
-#![feature(non_null_convenience)]
 #![feature(debug_closure_helpers)]
 #![feature(allocator_api)]
 #![feature(map_try_insert)]
@@ -10,10 +9,9 @@
 #![feature(get_many_mut)]
 #![feature(map_many_mut)]
 #![feature(trait_alias)]
-#![feature(pointer_is_aligned)]
+#![feature(pointer_is_aligned_to)]
 #![feature(buf_read_has_data_left)]
-#![feature(option_take_if)]
-#![feature(exclusive_range_pattern)]
+#![feature(iterator_try_collect)]
 
 mod db;
 mod os;

--- a/src/paging/cache.rs
+++ b/src/paging/cache.rs
@@ -395,15 +395,19 @@ impl Cache {
         &mut self,
         pages: [PageNumber; N],
     ) -> Option<[&mut MemPage; N]> {
-        self.pages.get_many_mut(pages.each_ref()).map(|frame_ids| {
-            self.buffer
-                .get_many_mut(frame_ids.map(|frame_id| *frame_id))
-                .unwrap()
-                .map(|frame| {
+        self.buffer
+            .get_many_mut({
+                self.pages
+                    .get_many_mut(pages.each_ref())
+                    .map(|frame_id| *frame_id.unwrap())
+            })
+            .ok()
+            .map(|frames| {
+                frames.map(|frame| {
                     frame.set(REF_FLAG | DIRTY_FLAG);
                     &mut frame.page
                 })
-        })
+            })
     }
 
     /// Returns the [`FrameId`] of `page_number` and automatically sets its ref

